### PR TITLE
Clean up BatchWorker

### PR DESCRIPTION
**Pull request by ReubenBond**
No functional change, just a minor cleanup and minor reduction of API surface

### CODE REVIEW

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

File: src/Orleans.Core/Async/BatchWorker.cs [hunk 1]
```diff
@@ -13,15 +13,29 @@ namespace Orleans
...
 
+        // Used to communicate the task for the next work cycle to waiters.
+        // This value is non-null only if there are waiters.
+        private TaskCompletionSource<Task> nextWorkCyclePromise;
+
+        /// <summary>
+        /// Creates a new worker from the provided delegate.
+        /// </summary>
+        public static BatchWorker Create(Func<Task> work) => new BatchWorkerFromDelegate(work);
+
```
[the remark this answers is not in the document]
Comment (jason-bragg) [reply src/Orleans.Core/Async/BatchWorker.cs]: > Given there's only one implementation of this type, we could flatten the hierarchy and make this class non-abstract.

I'm ok with flattening it.  I'm unclear why we've an abstraction and if we need one, we can always add one back in.  If we do have an abstraction, it should be a clean abstraction, which it was prior to this change.

[the remark this answers is not in the document]
Comment (jason-bragg) [reply src/Orleans.Core/Async/BatchWorker.cs]: Regarding the public abstraction being part of Orleans.  I agree.  We can move them and share the code as we do with the storage utilities, but that doesn't need to be part of this cleanup, unless you're feeling energetic. :)

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Pull request merged**